### PR TITLE
Enhance Create Recipe Screens UI and functionality

### PR DESCRIPTION
**Pull request by 2Fick**
## Description
In this PR, I’m working on improving the screens involved in the recipe creation process. 

### `RecipeNameScreen`

- **Error Message Visibility**: The error message was previously hidden behind the "Next Step" button. I’ve adjusted its position to ensure it’s fully visible and readable. 
 **Before**             |  **After**
    :-------------------------:|:-------------------------:
    image  |  image

- **Recipe Name Initialization**: Updated the `recipeName` initialization to pull the value from `CreateRecipeViewModel`. This allows the field to retain the input when navigating back to this screen.  
  ```kotlin
  // Updated recipeName initialization
  var recipeName by remember { 
      mutableStateOf(TextFieldValue(createRecipeViewModel.getRecipeName())) 
  }
  ```

- **Title Font Update**: Changed the font of the title text to enhance readability and improve alignment with the overall design.
**Before**             |  **After**
    :-------------------------:|:-------------------------:
    image  |  image  

- **Chef Image Quality**: Replaced the `ChefImage` with a higher-quality version to improve visual clarity and overall presentation.  
 **Before**             |  **After**
    :-------------------------:|:-------------------------:
    image  |  image
 

- **Improved Recipe Name Input with Placeholder**: Previously, the `label` was used in the `OutlinedTextField` for entering the recipe name, which resulted in the label text appearing above the field even when typing. This cluttered the UI and misaligned the text within the input. Now, I've replaced the `label` with a `placeholder`, which is displayed inside the field and disappears when the user starts typing. This adjustment improves text alignment and creates a cleaner look.

  _**Before** (using label):_  
  image

  _**After** (using placeholder):_  
  
  image
  image

### `MainActivity`: 

- **Refactored currentStep Values**: Following review feedback, I replaced hardcoded `currentStep` values with constants across the `MainActivity` composables to improve code readability and maintainability. This change makes it easier to understand the specific steps within the recipe creation flow.

  - `RecipeIngredientsScreen` now uses the constant `SECOND_STEP_OF_THE_CREATION`.
  - `RecipeInstructionsScreen` now uses the constant `THIRD_STEP_OF_THE_CREATION`.

This refactoring ensures that navigation steps are more descriptive and easier to manage in the future.

### `RecipeStepScreen`

- **Updated Typography and Image Quality in RecipeStepScreen**: Applied the same improvements made in `RecipeNameScreen` to ensure consistency across the recipe creation flow:
  - Changed the typography of the title text to match the updated styling in `RecipeNameScreen`.
  - Replaced `ChefImage` with a higher-quality version for better visual clarity.

These changes can be observed in the following screens:
  - `RecipeIngredientsScreen`
  - `RecipeInstructionsScreen`

### `IconDropdownMenu`:
- **Localization and Consistency**: Updated `IconType` to use `descriptionResId` (resource IDs) instead of hardcoded strings for descriptions, allowing `stringResource` to handle localization.
- **Consistent Border and Text Colors**: Aligned border color and adjusted the "Add Icon" placeholder text color to match other input fields for a consistent look.
- **Icon Selection Display**: When an icon is selected, its description is now displayed next to the icon, enhancing clarity for the user.

### `AddInstructionStepScreen`:
- **Category Field Removal**: Removed the category input field, as it’s currently unnecessary for this screen.
- **Consistent Border Styling**: Matched the border color of text fields with the `IconDropdownMenu` for a unified appearance.
- **Instruction Field Updates**: Set a minimum and maximum height for the Instruction field using `minLines`, making it scrollable if the content exceeds the visible area, to accommodate longer descriptions.
image

### `PublishRecipeScreen`:
  - **Signature Update**: Modified the function signature to include `userViewModel`, allowing access in the `NavHost` setup within `MainActivity`. Now, when navigating to `Screen.PUBLISH_CREATED_RECIPE`, `PublishRecipeScreen` can access both view models:
    ```kotlin
    composable(Screen.PUBLISH_CREATED_RECIPE) {
        PublishRecipeScreen(
            navigationActions = navigationActions,
            createRecipeViewModel = createRecipeViewModel,
            userViewModel = userViewModel
        )
    }
    ```
  - **Chef Image Update**:  Replaced the chef image with a higher-quality version. This update ensures that when the image size is adjusted on larger screens, the quality remains clearer and degrades less than with the previous image. Below is a before-and-after comparison of the chef image:

    **Before**             |  **After**
    :-------------------------:|:-------------------------:
    image  |  image

  - **Recipe Publishing Enhancement**: Integrated functionality where a recipe, once successfully published to the database, is also added to the user’s created recipes list within `userViewModel`.
  - **Testing Improvements**: Added a test tag to the publish button for more precise targeting in UI tests.

### `CreateRecipeViewModel`:
  - Updated the `publishRecipe` function to support success and failure callbacks, allowing `PublishRecipeScreen` to handle these states directly:
    ```kotlin
    fun publishRecipe(
        onSuccess: (Recipe) -> Unit,
        onFailure: (Exception) -> Unit
    ) { 
        // implementation
    }
    ```

### Test Updates:
- Adjusted `CreateRecipeViewModelTest` and `PublishRecipeScreenTest` to verify the updated behavior, ensuring that the recipe is correctly published and added to the user's created recipes list in success scenarios, with error handling on failure.

- **AddInstructionStepScreenTest**: Updated test cases to reflect the recent changes in both `IconDropdownMenu` and `AddInstructionStepScreen`, ensuring all tests align with the modified UI and functionalities.

## Checklist
- [x] Tests added.
- [x] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #138

### CODE REVIEW

File: app/src/main/java/com/android/sample/MainActivity.kt
Comment (akshire) [inline app/src/main/java/com/android/sample/MainActivity.kt:1]: LGTM!

File: app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt
Comment (akshire) [inline app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt:1]: LGTM!

File: app/src/main/java/com/android/sample/ui/createRecipe/AddInstructionStepScreen.kt
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/AddInstructionStepScreen.kt:1]: Great work on this screen, when testing the UI I immediately saw the improvement !!

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeStepScreen.kt
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/RecipeStepScreen.kt:1]: Great Work also on this screen. Incredible improvements in the Image quality!! :)

**Review by akshire — COMMENTED** [review]
Hey great PR with a lot of enhancements on the Create Recipe Screens. Big kudos to you!! I have tested the interface and it's great. I can see that you put a lot of effort in it. 

There are a lots of comments but don't be afraid, they are mainly reminders to put magic values in the `C` file

File: app/src/main/java/com/android/sample/ui/createRecipe/IconDropdownMenu.kt [hunk 1]
```diff
@@ -43,29 +43,39 @@ fun IconDropdownMenu(
...
+                    modifier = Modifier.weight(1f))
               } else {
+                Spacer(modifier = Modifier.weight(0.05f))
                 Text(
                     text = stringResource(R.string.add_icon),
                     style = Typography.bodyMedium,
+                    color = Color.DarkGray,
                     modifier = Modifier.weight(1f))
               }
               Spacer(modifier = Modifier.width(4.dp))
```
[the remark this answers is not in the document]
Comment (2Fick) [reply app/src/main/java/com/android/sample/ui/createRecipe/IconDropdownMenu.kt:75]: I used the MaterialTheme.colorScheme.onSecondary instead for consistency

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeStepScreen.kt [hunk 2]
```diff
@@ -72,34 +68,30 @@ fun RecipeStepScreen(
                     style = MaterialTheme.typography.bodyMedium,
                     color = MaterialTheme.colorScheme.onPrimary,
                     modifier =
-                        Modifier.padding(horizontal = BASE_PADDING * 2).width(260.dp).height(63.dp),
+                        Modifier.padding(horizontal = BASE_PADDING * 2)
+                            .width(260.dp)
+                            .height(63.dp)
```
[the remark this answers is not in the document]
Comment (2Fick) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeStepScreen.kt:73]: I deleted it as they are no real need to keep them

**Review by akshire — APPROVED** [review]
LGTM!!

**Pull request merged**